test/tchannel_bridge.js

```diff
@@ -0,0 +1,122 @@
+// @flow
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import _ from 'lodash';
+import {assert} from 'chai';
+import * as constants from '../src/constants';
+import ConstSampler from '../src/samplers/const_sampler';
+import path from 'path';
+import InMemoryReporter from '../src/reporters/in_memory_reporter';
+import opentracing from 'opentracing';
+import Tracer from '../src/tracer';
+import TChannel from 'tchannel';
+import TChannelBridge from '../src/tchannel_bridge.js';
+import TChannelAsThrift from 'tchannel/as/thrift';
+import TChannelAsJSON from 'tchannel/as/json';
+
+describe ('test tchannel span bridge', () => {
+    // BIG_TIMEOUT is useful for debugging purposes.
+    let BIG_TIMEOUT = 15000000;
+    let tracer = new Tracer(
+        'test-service',
+        new InMemoryReporter(),
```

Comment:
this test must verify that the right spans are reported. Right now you never finish spans